Subtype Parser

jovianyu opened the pull request:
Initial script for adding streamlining subtype information as well as
core events, with test input.

Commit:
Subtype Parser

Initial script for adding streamlining subtype information as well as
core events, with test input.

mdml commented on subtype-parser/subtype.py:
```diff
@@ -0,0 +1,67 @@
+    with open( tempDirectory + 'temp.snv', "w" ) as out:
+        output = []
+        for p in MM:
+            if subtypeDict[p[0]] != None:
```
This is not quite right. If patient _P_ is of subtype _T_, we want to add events (aberrations) for all subtypes **except** _T_. That way genes with alterations that are mutually exclusive with the subtype _T_ alteration are actually enriched for mutations in _T_.

bournewu replied:
Line 59 should change to only adding subtypes that are not in the patient P. For example, 

``` python
p.extend( [ s for s in subtypes.difference( set([subtypeDict[p[0]]]) ) ])
```

Commit:
Adding sample inputs

Sample inputs.

Commit:
Fixed subtype inclusion criteria

Commit:
Adding in subtype information for samples without subtype info

Commit:
Subtype parser integration

Commit:
Subtype parser fully integrated

Fully integrated and verified on STAD gastric data:
http://compbio-research.cs.brown.edu/comet-test/

jovianyu commented:
Subtype parser now fully merged into the CoMEt codebase and verified on test data.

mdml commented on comet/mutation_data.py:
```diff
@@ -20,31 +20,49 @@ def load_mutation_data(filename, patientFile=None, geneFile=None, minFreq=0):
             patient, mutations = arr[0], set(arr[1:])
 
+            if subtypeDict[patient]:
+                mutations = mutations.union( subtypes.difference( set( [subtypeDict[patient]] ) ) )
```
I would write something like the following instead:

```
mutations |= set( s for s in subtypes if subtypeDict[patient] != s )
```

mdml commented on comet/mutation_data.py:
```diff
@@ -20,31 +20,49 @@ def load_mutation_data(filename, patientFile=None, geneFile=None, minFreq=0):
             if not patients or patient in patients:
                 if genes: mutations &= genes
-                #else: genes |= mutations                
+                #else: genes |= mutations
```
Why is this commented out? Should it be deleted?

mdml commented on run_comet.py:
```diff
@@ -246,14 +250,15 @@ def run( args ):
-    if args.subtype:
-        with open(args.subtype) as f:
-            subSet = [ l.rstrip() for l in f ]
+    if args.core_events:
```
This looks good, but can you add some comments to explain the logic here?

mdml commented:
This looks good to me once you address the few comments I made.

@bournewu: can you review and test Jovian's pull request?

Commit:
Mutation data and subtype processing update

- Changed a line with feedback from @mdml
- Removed an old comment line that doesn’t seem to be doing
anything/necessary
- Added comments in run_comet.py

bournewu commented:
Just reviewed and tested. Now looks good to me. @jovianyu could you merge it?

Just merged to the master. : )
